### PR TITLE
gltf export: Fix export of skeletons and skins.

### PR DESCRIPTION
**Pull request by lyuma**
cc @fire
Create GLTFSkeleton at the same time we create GLTFNode objects.
Create GLTFSkin at the same time we export MeshInstance3D
Also, convert BoneAttachment3D nodes during normal node creation.

This avoids needing special cases during mesh export, and especially exporting skeletons or meshes which are children of BoneAttachment3D.

Should address some of issue #50020. I have not verified the fix, but the old code which dealt with Skeleton or Skin export was very broken so this should be a starting point for future GLTF fixes related to animated or skinned meshes.

This is still in draft state because some of my meshes with blend shapes (but not all of them) fail to load in Blender
[Blender error "index 990 is out of bounds for axis 0 with size 990"](https://user-images.githubusercontent.com/39946030/134859065-362a4092-867a-497a-a6a8-a9d6469f717b.png)

Other test cases include the reproduction case I call "Party Hat" since it has a skinned "Party Hat" with springbones attached to a BoneAttachment3D onto the Head bone of a character's Skeleton3D. It seemed to previously reproduce interesting issues on export.

There are a few other glTF files in my collection of cursed glTF models which I still need to test re-exporting, containing duplicate skins, nested skeletons (including cases where a mesh is its own bone), and so on.

### CODE REVIEW

File: modules/gltf/gltf_document.h [hunk 1]
```diff
@@ -369,20 +374,27 @@ class GLTFDocument : public Resource {
...
 			const GLTFAnimationIndex index, const int bake_fps);
-	GLTFMeshIndex _convert_mesh_instance(Ref<GLTFState> state,
-			MeshInstance3D *p_mesh_instance);
 	void _convert_mesh_instances(Ref<GLTFState> state);
 	GLTFCameraIndex _convert_camera(Ref<GLTFState> state, Camera3D *p_camera);
-	void _convert_light_to_gltf(Light3D *light, Ref<GLTFState> state, Node3D *spatial, Ref<GLTFNode> gltf_node);
+	void _convert_light_to_gltf(Light3D *light, Ref<GLTFState> state, Ref<GLTFNode> gltf_node);
 	GLTFLightIndex _convert_light(Ref<GLTFState> state, Light3D *p_light);
-	GLTFSkeletonIndex _convert_skeleton(Ref<GLTFState> state, Skeleton3D *p_skeleton);
+	/*GLTFSkeletonIndex _convert_skeleton(Ref<GLTFState> state, Skeleton3D *p_skeleton) {
```
Comment (fire) [inline modules/gltf/gltf_document.h:381]: Code that isn't used should be deleted.

**Comment by fire** [comment]
Some minor problems with variable shadowing and documentation has been modified causing CICD to break.

**Comment by fire** [comment]
Exporting blend shapes as not reset to default causes the face blendshapes to explode.

**Comment by lyuma** [comment]
I have made some additional fixes to blendshapes. They should no longer have such glitches, hopefully.

This should now be in a mergable state.

**Review by fire — APPROVED** [review]
Looks good to me. Passes all the CICD tests.

* Blend shapes export and https://github.com/lyuma/godot-tests/tree/add_glb worked. 
* Passes my double fox test. Which is a scene with two fox.glbs instanced.

**Comment by akien-mga** [comment]
Thanks!
